lib/llm/src/discovery/model_manager.rs

```diff
@@ -310,23 +316,38 @@ impl ModelManager {
         }
 
         let client = endpoint.client().await?;
-        let store = endpoint.component().drt().store();
-        let router_bucket = store
-            .get_or_create_bucket(KV_ROUTERS_ROOT_PATH, None)
-            .await?;
-        let router_uuid = uuid::Uuid::new_v4();
-        // In lib/llm/src/kv_router/subscriber.rs we filter on component.service_name() so this
-        // must have that prefix.
-        let router_key = Key::new(format!(
-            "{}/{}/{}",
-            endpoint.component().service_name(),
-            endpoint.name(),
-            router_uuid,
-        ));
-        let json_router_config = serde_json::to_vec_pretty(&kv_router_config.unwrap_or_default())?;
-        router_bucket
-            .insert(&router_key, json_router_config.into(), 0)
-            .await?;
+
+        // Register router via discovery mechanism
+        let discovery = endpoint.component().drt().discovery();
+        let instance_id = discovery.instance_id();
+
+        // Build NATS transport subject for the router endpoint
+        // Use KV_ROUTER_COMPONENT as the component name to distinguish from the generate endpoint's component
+        let router_endpoint_id = EndpointId {
+            namespace: endpoint.component().namespace().name().to_string(),
+            component: KV_ROUTER_COMPONENT.to_string(),
+            name: KV_ROUTER_ENDPOINT.to_string(),
+        };
+        // Placeholder subject - router is not callable, only registered for lifecycle coordination
+        let nats_subject = format!(
```

Comment:
question: this nats subject is used for router synchronization? 

it's not for calling the router endpoints - is that correct?

Could this be guarded by if router syncronization / snapshotting / kv events (not sure which capabilities specifically) is enabled?